packages/desktop-client/src/components/util/AmountInput.js

```diff
@@ -59,6 +59,15 @@ export function AmountInput({
     }
   }
 
+  // Save changes when the component unmounts.
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm afraid this won't be that simple.

Currently `fireChange` will trigger whenever `value !== initialValue`. Not just on unmount. So you'll be triggering much more events than expected.

What you actually want is to trigger the `onBlur` event when unmounting the `Input` component. But that's not really straight-forward.

See the thread here: https://github.com/facebook/react/issues/12363

Reply:
I read that issue and tried the `display: none` solution. It worked on other parts of the codebase but for some reason when used here, the `value` state is always zero.